### DISCUSSION TITLE
Homework_00

### DISCUSSION THREAD

**Discussion by C-Olsen** [cooperrc/advanced-dynamics#12] [Q&A] [answered]
Two questions regarding Homework_00.  For the professor or anyone in the class who has the answers

1.  What format should python based homework assignments be submitted in?  Should we submit Juptyer notebook file format?  Are individual file submissions expected for each problem or condensed to a single notebook?

<s>2. Homework_00 has problems 1, 2, & 3.  Problem 3 does not appear to have a challenge problem or question asked.  Additionally the supplied code does not run in Jupyter.  The error is also present on the class assignment page.  What is the expectation for Problem 3?</s>

On further review I realize I was looking at the example immediately following problem 3, rather than problem 3 itself.  Please disregard the second question.

**Comment by erooker** [comment, marked as answer]
For the file format he mentioned submitting as pdf in this video https://youtu.be/cIsFmVVtmYE?si=q5F4b8wO2WWh8vOE

**Reply by C-Olsen** [reply]
Thank you!

**Reply by travismalave** [reply]
Can you share where you found this YouTube link? I wasn't able to find anything linking to this video in the syllabus or HuskyCT and I'm wondering if I'm missing something.

**Comment by tad12002** [comment]
Also I dont know if this matters but he printed a html preview before downloading as a pdf. There is an option to download as a pdf directly I dont know if there is difference to the order he did in the video but I assume there is.

**Comment by travismalave** [comment]
To comment on the last portion of HW00 - the code works after fixing a quick typo. In the code when defining init(): the next line calls out for "line.set_data...." but "line isn't defined. Simply add a "1" after line since the code refers to line1 and line2 and the plot works out. 

Don't think this is part of the assignment but more of an FYI.

**Reply by C-Olsen** [reply]
I caught the variable name error.  My issue was not running (shift+enter) the previous line of code where F, dx, and W were defined.  Passing that along if anyone is still stuck.

Thanks!

**Comment by HuyNgo24** [comment]
You can take a look the comment below for fixing the error
https://github.com/cooperrc/advanced-dynamics/issues/9#issue-2084696650

or you can use these updated code:

```
import matplotlib.pyplot as plt

from matplotlib import animation, rc
from IPython.display import HTML

fig, ax = plt.subplots()

ax.set_xlim(( 0, 3))
ax.set_ylim((-0.1, 5))

line1, = ax.plot([], [], 's')
line2, = ax.plot([], [], 'r-v')

def init():
    line1.set_data([], [])
    line2.set_data([],[])
    return (line1, line2)
    
def animate(i):
    x = np.linspace(0, 1, 100)
    line1.set_data(x[i], 0)
    line2.set_data([x[i], x[i] + F[0]], [0, 0 + F[1]]) 
    #plt.quiver(x[i],0,F[0],F[1])
    return (line1, line2)
    
anim = animation.FuncAnimation(fig, animate, init_func=init,
                               frames=100, interval=20, blit=True)

HTML(anim.to_html5_video())
```

**Reply by C-Olsen** [reply]
Thanks!
